website/src/components/FlaggableElement.tsx

```diff
@@ -0,0 +1,205 @@
+import poster from "src/lib/poster";
+import {
+  Popover,
+  useBoolean,
+  PopoverAnchor,
+  Button,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  Checkbox,
+  PopoverCloseButton,
+  PopoverArrow,
+  Flex,
+  Spacer,
+  Slider,
+  SliderTrack,
+  SliderFilledTrack,
+  SliderThumb,
+} from "@chakra-ui/react";
+import useSWRMutation from "swr/mutation";
+import { QuestionMarkCircleIcon, FlagIcon } from "@heroicons/react/20/solid";
+import { useState } from "react";
+
+export type TextFlagProps = {
+  text: string;
+  label_map;
+  post_id: string;
+};
+type FlagRefs = {
+  label: string;
+  value: number;
+  setValue;
+  isChecked: boolean;
+  setChecked;
+};
+
+export const FlaggableElement = (props) => {
+  const [isEditing, setIsEditing] = useBoolean();
+  const [hasCheck, setHasCheck] = useBoolean();
+  const { trigger, isMutating } = useSWRMutation("/api/v1/text_labels", poster, {
+    onSuccess: () => {
+      setIsEditing.off;
+    },
+  });
+
+  const submitResponse = (label_data) => {
+    trigger(label_data);
+  };
+  const label_refs: FlagRefs[] = TEXT_LABEL_FLAGS.map((flag) => {
```

Comment:
this works now just fine because the flags are all constant, but in-case the number of flags is dynamic (i.e. comes from the server) this will error because of different number of hook calls.

Nothing to do here, just keep in mind for the future.

Reply:
thanks for the input! thought i was being clever, but yes I have changed to useState on array